src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -162,6 +163,11 @@ public override void Load()
 
         public async Task RefreshAsync(CancellationToken cancellationToken)
         {
+            if (!_configClientManager.HasAvailableClients)

```

Comment:
Can we move this to the inside of the interlock so we don't check time like crazy?

Reply:
Done!